src/gsi/read_satwnd.f90

```diff
@@ -1002,327 +1141,208 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
-!                 ! THIS CHANGE HAS NOT BEEN TESTED !!!
-!                 !call ufbrep(lunin,amvcld,12,irep_array,iret, 'FOST CDTP MUCE
-!                 VSAT TMDBST VSAT CDTP MUCE OECS CDTP HOCT COPT')
-!                 call ufbrep(lunin,amvcld,12,irep_array,iret, 'FOST CDTP MUNCEX
-!                 VSAT TMDBST VSAT CDTP MUNCEX OECS CDTP HOCT COPT')
-!                 deallocate( amvcld )
-
-                 call ufbseq(lunin,amvqic,2,4,iret, 'AMVQIC') ! AMVQIC:: GNAPS PCCF
-                 qifn = amvqic(2,2)  ! QI w/ fcst does not exist in this BUFR
-                 ee = amvqic(2,4) ! NOTE: GOES-R's ee is in [m/s]
-              endif
+!              call ufbint(lunin,rep_array,1,1,iret, '{AMVIII}')
+!              irep_array = int(rep_array)
+!              allocate( amviii(12,irep_array))
+!              call ufbrep(lunin,amviii,12,irep_array,iret, 'LTDS SCLF SAID
+!              SIID CHNM SCCF ORBN SAZA BEARAZ EHAM PRLC TMDBST')
+!              deallocate( amviii )
+
+              call ufbint(lunin,rep_array,1,1,iret, '{AMVIVR}')
+              irep_array = int(rep_array)
+              allocate( amvivr(2,irep_array))
+              call ufbrep(lunin,amvivr,2,irep_array,iret, 'TCOV CVWD')
+              pct1 = amvivr(2,1)     ! use of pct1 (a new variable in the BUFR) is introduced by Nebuda/Genkova
+              deallocate( amvivr )
+
+!              call ufbrep(lunin,rep_array,1,1,iret, '{AMVCLD}')
+!              irep_array = int(rep_array)
+!              allocate( amvcld(12,irep_array))
+!              ! MUCE --> MUNCEX within the new GOES16/17 and NOAA-20 VIIRS
+!              sequence (I.Genkova, J.Whiting)
+!              ! THIS CHANGE HAS NOT BEEN TESTED !!!
+!              !call ufbrep(lunin,amvcld,12,irep_array,iret, 'FOST CDTP MUCE
+!              VSAT TMDBST VSAT CDTP MUCE OECS CDTP HOCT COPT')
+!              call ufbrep(lunin,amvcld,12,irep_array,iret, 'FOST CDTP MUNCEX
+!              VSAT TMDBST VSAT CDTP MUNCEX OECS CDTP HOCT COPT')
+!              deallocate( amvcld )
+
+              call ufbseq(lunin,amvqic,2,4,iret, 'AMVQIC') ! AMVQIC:: GNAPS PCCF
+              qifn = amvqic(2,2)  ! QI w/ fcst does not exist in this BUFR
+              ee = amvqic(2,4) ! NOTE: GOES-R's ee is in [m/s]
 ! Extra block for VIIRS NOAA20: End
 ! Extra block for GOES-R winds: Start
-           else if(trim(subset) == 'NC005030' .or. trim(subset) == 'NC005031' .or. trim(subset) == 'NC005032' .or. &  !IR(LW) / CS WV / VIS  GOES-R like winds
-                   trim(subset) == 'NC005034' .or. trim(subset) == 'NC005039' .or. trim(subset) == 'NC005099') then   !CT WV  / IR(SW) GOES-R like winds
+           else if (istype >= 15 .and. istype <=20)then
 
-              if ( trim(subset) == 'NC005099' ) then
-                hdrdat(10)=61.23 ! set zenith angle for CIMSS AMVs to 67 to pass QC, no value in origional data
-              end if
-              if(hdrdat(1) >=r250 .and. hdrdat(1) <=r299 ) then  ! the range of NESDIS satellite IDs
```

Comment:
Thank you for your thorough review.  I will have to work through your comments one by one over the next few days.

Reply:
I deleted this line because in the new code when the itype is defined, the check is done.  So for those data that do not pass, itype is equal to zero and the data is not used.  (I think I did things correctly).